pkg/kubelet/lifecycle/handlers.go

```diff
@@ -113,21 +121,46 @@ func (hr *HandlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 		}
 		host = status.IP
 	}
-	var port int
-	if handler.HTTPGet.Port.Type == intstr.String && len(handler.HTTPGet.Port.StrVal) == 0 {
-		port = 80
-	} else {
-		var err error
-		port, err = resolvePort(handler.HTTPGet.Port, container)
-		if err != nil {
-			return "", err
-		}
+	port, err := resolvePort(handler.HTTPGet.Port, container)
+	if err != nil {
+		return "", err
 	}
 	url := fmt.Sprintf("http://%s/%s", net.JoinHostPort(host, strconv.Itoa(port)), handler.HTTPGet.Path)
 	resp, err := hr.httpGetter.Get(url)
 	return getHttpRespBody(resp), err
 }
 
+func (hr *HandlerRunner) runSocketHandler(pod *v1.Pod, container *v1.Container, handler *v1.Handler) (string, error) {
+	host := handler.TCPSocket.Host
+	// default to pod ip
+	if len(host) == 0 {
+		status, err := hr.containerManager.GetPodStatus(pod.UID, pod.Name, pod.Namespace)
+		if err != nil {
+			glog.Errorf("Unable to get pod info, event handlers may be invalid.")
+			return "", err
+		}
+		if status.IP == "" {
+			return "", fmt.Errorf("failed to find networking container: %v", status)
+		}
+		host = status.IP
+	}
+	port, err := resolvePort(handler.HTTPGet.Port, container)
+	if err != nil {
+		return "", err
+	}
+
+	addr := net.JoinHostPort(host, strconv.Itoa(port))
+	conn, err := net.DialTimeout("tcp", addr, 10*time.Second)
```

Comment:
Can you deduplicate this with [DoTCPProbe](https://github.com/kubernetes/kubernetes/blob/152b0c12da40d85b087100b805cc348db613ee9d/pkg/probe/tcp/tcp.go#L47)